src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -726,6 +726,61 @@ theorem getLsb_sshiftRight (x : BitVec w) (s i : Nat) :
         Nat.not_lt, decide_eq_true_eq]
       omega
 
+/-! ### signExtend -/
+
+/-- Equation theorem for 'Int.sub' when both arguments are 'negSucc' -/
+private theorem Int.ofNat_sub_ofNat_of_lt {n m : Nat} (hlt : n < m) :
+    (n : Int) - (m : Int) = -(↑(m - 1 - n) + 1) := by
+  omega
+
+/-- Equation theorem for 'Int.mod' -/
+private theorem Int.negSucc_emod (m : Nat) (n : Int) :
+    -(m + 1) % n = Int.subNatNat (Int.natAbs n) ((m % Int.natAbs n) + 1) := rfl
+
+
+/-- To sign extend when the msb is false, then sign extension is the same as truncation -/
+theorem signExtend_eq_neg_truncate_neg_of_msb_false {x : BitVec w} {v : Nat} (hmsb : x.msb = false) :
+    (x.signExtend v) = x.zeroExtend v := by
+  ext i
+  by_cases hv : i < v
+  · simp only [getLsb_zeroExtend, hv, decide_True, Bool.true_and]
+    rw [signExtend, getLsb]
+    simp only [toNat_ofInt]
+    rw [BitVec.toInt_eq_msb_cond]
+    simp only [hmsb, Bool.false_eq_true, ↓reduceIte]
+    rw [Int.ofNat_mod_ofNat, Int.toNat_ofNat]
+    simp only [Nat.testBit_mod_two_pow, hv, decide_True, Bool.true_and]
+    rw [BitVec.testBit_toNat]
+  · simp only [getLsb_zeroExtend, hv, decide_False, Bool.false_and]
+    apply getLsb_ge
+    omega
+
+/-- To sign extend when the msb is true, we perform double negation to make the high bits true. -/
+theorem signExtend_eq_neg_truncate_neg_of_msb_true {x : BitVec w} {v : Nat} (hmsb : x.msb = true) :
+    (x.signExtend v) = ~~~((~~~x).zeroExtend v) := by
```

Comment:
`truncate` in the name vs `zeroExtend` in the statement?

Reply:
Thanks for catching this, @tobiasgrosser found `zeroExtend` more intuitive, but it slipped me to change it consistently everywhere else. Addressed now.